### PR TITLE
Implement game start

### PR DESCRIPTION
**Pull request by Dieheart007**
Initial implementation of Game Start with Main Menu and Fight Menu. Review needed for fine tuning.

### CODE REVIEW

File: src/main/java/spring/game/GameStart.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    private boolean attacking;
+
+    public GameStart() {
+        // Need to update this default "player"
+        this.player = new Entity("", "Player", 100, 50, 50);
+    }
+
+    // Prompt user for name
+    public void promptForName() {
+        Scanner input = new Scanner(System.in);
```
Comment (adamhl8) [inline src/main/java/spring/game/GameStart.java:20]: I think we can clean this up a bit by making the Scanner a field so we don't have to create a new one in each method.

File: src/main/java/spring/game/GameStart.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+                case 2:
+                    break;
+
+                default:
+                    System.out.println("Please enter a valid choice!\n");
+                    break;
+            }
+          // Delete choice != 1 once functionality is expanded
+        } while (this.player.getHealth() != 0 && choice != 2 );
```
Comment (adamhl8) [inline src/main/java/spring/game/GameStart.java:92]: The player might be able to get stuck in this loop if the health doesn't drop to exactly 0. I think this check should be `this.player.getHealth() > 0`.

**Review by adamhl8 — COMMENTED** [review]
Looks great. I had no idea you could do something like `System.setIn(new ByteArrayInputStream("Cloud".getBytes()));`. That's pretty cool.

File: src/main/java/spring/game/GameStart.java (hunk 2 above)
Comment (Dieheart007) [reply src/main/java/spring/game/GameStart.java:92]: Yup, good catch.

File: src/main/java/spring/game/GameStart.java (hunk 1 above)
Comment (Dieheart007) [reply src/main/java/spring/game/GameStart.java:20]: Agreed.

Comment (Dieheart007) [reply src/main/java/spring/game/GameStart.java:20]: Doing that breaks all my tests for some reason.  No clue.

File: src/main/java/spring/game/GameStart.java (hunk 2 above)
Comment (Dieheart007) [reply src/main/java/spring/game/GameStart.java:92]: Done.

File: src/main/java/spring/game/GameStart.java (hunk 1 above)
Comment (adamhl8) [reply src/main/java/spring/game/GameStart.java]: There's an issue when trying to run tests through gradle where you can't overwrite stdin. We can just ignore writing tests for the game logic for now.

**Pull request merged**